crengine/src/lvrend.cpp

```diff
@@ -3897,15 +3919,15 @@ int pagebreakhelper(ldomNode *enode,int width)
 {
     int flag=css_pb_auto;
     int em = enode->getFont()->getSize();
-    int margin_top = lengthToPx( enode->getStyle()->margin[2], width, em ) + DEBUG_TREE_DRAW;
-    int padding_top = lengthToPx( enode->getStyle()->padding[2], width, em ) + DEBUG_TREE_DRAW+measureBorder(enode,0);
+    int margin_top = lengthToPx( enode, enode->getStyle()->margin[2], width, em ) + DEBUG_TREE_DRAW;
+    int padding_top = lengthToPx( enode, enode->getStyle()->padding[2], width, em ) + DEBUG_TREE_DRAW+measureBorder(enode,0);
     flag=CssPageBreak2Flags(getPageBreakBefore(enode))<<RN_SPLIT_BEFORE;
     if (flag==RN_SPLIT_BEFORE_ALWAYS){
         ldomNode *node=enode;
         int top=0;
         while (!node->isNull()) {
-            top+=lengthToPx( node->getStyle()->margin[2], width, em ) +
-                 lengthToPx( node->getStyle()->padding[2], width, em ) +
+            top+=lengthToPx( enode, node->getStyle()->margin[2], width, em ) +
+                 lengthToPx( enode, node->getStyle()->padding[2], width, em ) +
```

Comment:
`node` (the parent we are iterating) instead of `enode` (the child we started from).
(Haven't checked all, just throwing this one that was obvious from the diff.)